src/main/kotlin/no/nb/bikube/core/controller/CoreController.kt

```diff
@@ -84,4 +92,41 @@ class CoreController (
             else -> throw NotSupportedException("Material type $materialType is not supported.")
         }
     }
+
+    @GetMapping("/item/search", produces = [MediaType.APPLICATION_JSON_VALUE])
+    @Operation(
+        summary = "Search catalogue items",
+        description = "Search catalogue items by title id, material type, date and digital/physical."
+    )
+    @ApiResponses(value = [
+        ApiResponse(responseCode = "200", description = "OK"),
+        ApiResponse(responseCode = "400", description = "Bad request"),
+        ApiResponse(responseCode = "500", description = "Server error")
+    ])
+    @Parameters(
+        Parameter(name = "titleCatalogueId", description = "Catalogue ID of the title to search items for."),
+        Parameter(name = "materialType", description = "Material type of the items to search for."),
+        Parameter(name = "date", description = "Date in ISO 8601 format (YYYY-MM-DD).", schema = Schema(pattern = DATE_REGEX)),
+        Parameter(name = "isDigital", description = "If 'true' returns digital items, physical items if 'false'. Default is 'false'")
+    )
+    fun searchItem(
+        @RequestParam(required = true) titleCatalogueId: String,
+        @RequestParam(required = true) materialType: MaterialType,
+        @RequestParam(required = true) date: String,
+        @RequestParam(required = true) isDigital: Boolean,
+    ): ResponseEntity<Flux<CatalogueRecord>> {
+        if (titleCatalogueId.isEmpty()) throw BadRequestBodyException("Search term cannot be empty.")
+
+        var parsedDate: LocalDate? = null
+        if (date.isNotEmpty()) parsedDate = runCatching { LocalDate.parse(date) }
+            .getOrElse { throw BadRequestBodyException("Date must valid ISO-8601 format") }
```

Comment:
If the `date` parameter is an empty string, `parsedDate` remains null and the subsequent call to `getItemsTitle` fails with NullPointerException, because of the non-null assertion `parsedDate!!`.

Suggestion:
```suggestion
        val parsedDate = date.takeIf { it.isNotEmpty() }
            ?.let {
                runCatching { LocalDate.parse(it) }
                    .getOrElse { throw BadRequestBodyException("Date must valid ISO-8601 format") }
            }
            ?: throw BadRequestBodyException("Date cannot be empty")
```

You can then remove the `!!` from the parameter list below.